core/blockly.js

```diff
@@ -392,21 +392,28 @@ Blockly.jsonInitFactory_ = function(jsonDef) {
  * @param {!Array.<!Object>} jsonArray An array of JSON block definitions.
  */
 Blockly.defineBlocksWithJsonArray = function(jsonArray) {
-  for (var i = 0, elem; elem = jsonArray[i]; i++) {
-    var typename = elem.type;
-    if (typename == null || typename === '') {
+  for (var i = 0; i < jsonArray.length; i++) {
```

Comment:
ooc, why change the way the loop is defined?

Reply:
Defining it the other way means it ends when it finds a null elem, even if it's not at the end of the actual list.

Reply:
where's the facepalm emoji when you need it?